content/3_Silver/Flood_Fill.mdx

```diff
@@ -796,3 +796,71 @@ print(room_num)
 ## Problems
 
 <Problems problems="general" hideSuggestProblemButton />
+
+## Quiz
+
+<Quiz>
+  <Quiz.Question>
+    In the context of flood fill algorithms for grids, what does the "visited" array primarily track?
+    <Quiz.Answer>
+      The number of cells in the grid.
+      <Quiz.Explanation>
+        Incorrect. The "visited" array is not used to count cells but rather to track which cells have already been visited to avoid re-processing.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer correct>
+      The cells that have been processed to avoid re-processing.
+      <Quiz.Explanation>
+        Correct! The "visited" array tracks cells that have been processed to prevent revisiting them during the flood fill algorithm.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      The adjacent cells for each node.
+      <Quiz.Explanation>
+        Incorrect. The "visited" array is not concerned with adjacent cells specifically but with tracking the processing status of each cell.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+  <Quiz.Question>
+    How does flood fill differ from depth-first search (DFS) in terms of its application in grid-based problems?
+    <Quiz.Answer correct>
+      Flood fill explores all nodes uniformly, while DFS may prioritize certain paths.
+      <Quiz.Explanation>
+        Correct! Flood fill ensures uniform exploration of connected nodes, while DFS may explore paths based on the order of traversal chosen.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      Flood fill does not require a stack for backtracking.
+      <Quiz.Explanation>
+        Incorrect. Flood fill, like DFS, typically uses a stack (or recursion) for backtracking to explore adjacent nodes.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      Flood fill only visits cells with the same value.
+      <Quiz.Explanation>
+        Incorrect. Flood fill explores all connected cells regardless of their values, aiming to label connected components.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+  <Quiz.Question>
+    What is the primary advantage of using a non-recursive implementation of flood fill over a recursive one?
```

Comment:
i dont think we should encourage them to use a non-recursive flood fill..

Reply:
yeah ok